rcl/test/rcl/test_events.cpp

```diff
@@ -215,20 +306,83 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription
     });
     ret = rcl_take(&subscription, &msg, nullptr);
     ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-    ASSERT_EQ(std::string(test_string), std::string(msg.string_value.data, msg.string_value.size));
+    EXPECT_EQ(std::string(test_string), std::string(msg.string_value.data, msg.string_value.size));
   }
 
+  ASSERT_TRUE(subscription_event_ready);
   if (subscription_event_ready) {
-    rmw_liveliness_changed_status_t liveliness_status;
-    ret = rcl_take_event(&subscription_event, &liveliness_status);
+    rmw_requested_deadline_missed_status_t deadline_status;
+    ret = rcl_take_event(&subscription_event, &deadline_status);
     ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-    // test liveliness_status contents
+    EXPECT_EQ(deadline_status.total_count, 1);
+    EXPECT_EQ(deadline_status.total_count_change, 1);
   }
 
+  ASSERT_TRUE(publisher_event_ready);
   if (publisher_event_ready) {
-    rmw_liveliness_lost_t liveliness_status;
-    ret = rcl_take_event(&publisher_event, &liveliness_status);
+    rmw_offered_deadline_missed_t deadline_status;
+    ret = rcl_take_event(&publisher_event, &deadline_status);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    EXPECT_EQ(deadline_status.total_count, 1);
+    EXPECT_EQ(deadline_status.total_count_change, 1);
+  }
+}
+
+/*
+ * Test of publisher and subscriber, expecting no deadline events
+ */
+TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_no_deadline_missed)
+{
+  SetUpPubSub(RCL_PUBLISHER_DEADLINE, RCL_SUBSCRIPTION_DEADLINE);
+
+  // TODO(wjwwood): add logic to wait for the connection to be established
+  //                probably using the count_subscriptions busy wait mechanism
+  //                until then we will sleep for a short period of time
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+  rcl_ret_t ret;
+
+  const char * test_string = "testing";
+  {
+    test_msgs__msg__Primitives msg;
+    test_msgs__msg__Primitives__init(&msg);
+    ASSERT_TRUE(rosidl_generator_c__String__assign(&msg.string_value, test_string));
+    ret = rcl_publish(&publisher, &msg);
+    test_msgs__msg__Primitives__fini(&msg);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  }
+
+  bool msg_ready, subscription_event_ready, publisher_event_ready;
+  ASSERT_TRUE(wait_for_msgs_and_events(&subscription, &subscription_event,
+    &publisher_event, context_ptr, 1, 1000, msg_ready, subscription_event_ready,
```

Comment:
I am still kind of confused here about why the wait doesn't timeout when we're correct about not expecting any deadline events to occur.

Reply:
Never mind. Found the bug.

However, after fixing the bug, the test is now failing and needs investigation.

Reply:
The bug is being tracked here https://github.com/aws-robotics/rmw_connext/issues/3.